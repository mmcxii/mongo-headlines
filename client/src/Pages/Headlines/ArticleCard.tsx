import React from 'react';
import styled from 'styled-components';

import { ArticleProps } from './Headlines';

interface Props {
    article: ArticleProps;
}

const ArticleCard: React.FC<Props> = ({ article }) => {
    return <p>{article.title}</p>;
};

export default ArticleCard;
