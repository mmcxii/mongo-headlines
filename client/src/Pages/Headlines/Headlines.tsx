import React, { useEffect, useState } from 'react';

import { useScrollToTop } from 'Hooks';
import ArticleCard from './ArticleCard';
import Pagination from './Pagination';

export interface ArticleProps {
    _id: string;
    title: string;
    abstract: string;
    updated_date: Date;
    url: string;
}

export interface CommentProps {
    _id: string;
    article: string;
    user: string;
    message: string;
    postedAt: Date;
}

interface Props {}

const Headlines: React.FC<Props> = () => {
    const [dataIsLoading, setDataIsLoading] = useState<boolean>(false);
    const [articles, setArticles] = useState<ArticleProps[]>([]);
    const [currentPage, setCurrentPage] = useState<number>(1);

    useScrollToTop(currentPage);

    useEffect(() => {
        const getData = async () => {
            setDataIsLoading(true);

            try {
                const res: Response = await fetch('/api/articles');
                const data: [] = await res.json();

                setArticles(data);
                setDataIsLoading(false);
            } catch (err) {
                console.log(err);
            }
        };

        getData();
    }, []);

    //* Pagination
    const indexOfLastArticle = currentPage * 10;
    const indexOfFirstArticle = indexOfLastArticle - 10;
    const currentArticles = articles.slice(indexOfFirstArticle, indexOfLastArticle);

    return (
        <>
            <section>
                <h2>Current Headlines</h2>
                {dataIsLoading ? (
                    <p>Loading...</p>
                ) : (
                    articles.length > 0 &&
                    currentArticles.map(article => <ArticleCard key={article._id} article={article} />)
                )}
            </section>

            <Pagination
                currentPage={currentPage}
                numberOfArticles={articles.length}
                setCurrentPage={setCurrentPage}
            />
        </>
    );
};

export default Headlines;
