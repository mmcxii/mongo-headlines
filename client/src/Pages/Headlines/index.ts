import ArticleCard from './ArticleCard';
import Headlines from './Headlines';

export * from './Headlines';
export { ArticleCard };

export default Headlines;
