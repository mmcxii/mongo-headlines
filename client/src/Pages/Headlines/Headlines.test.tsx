import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Headlines from './Headlines';

afterEach(cleanup);

test('<Headlines /> tests', () => {
    //* Arrange
    // @ts-ignore
    global.scrollTo = jest.fn();
    window.scrollTo = jest.fn();

    //* Act
    const { container } = render(<Headlines />);
    const component = container.firstChild;

    //* Assert
    expect(component).toMatchSnapshot();
});
