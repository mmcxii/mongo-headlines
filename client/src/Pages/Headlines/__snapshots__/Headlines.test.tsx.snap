// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headlines /> tests 1`] = `
<section>
  <h2>
    Current Headlines
  </h2>
  <p>
    Loading...
  </p>
</section>
`;
